tenacity/__init__.py

```diff
@@ -80,6 +82,15 @@
 from .after import after_nothing  # noqa
 
 
+def _waiter_takes_last_result(waiter):
+    if not six.callable(waiter):
+        return False
+    if isinstance(waiter, _wait.wait_base):
+        waiter = waiter.__call__
```

Comment:
Not sure that's gonna work for code that inherits from `wait_base` but is not updated.

Reply:
Seems like it should work?

```
>>> from tenacity import wait
>>> class OldWait(wait.wait_base):
...    def __call__(self, attempt, delay):
...        pass
...
>>> my_wait = OldWait()
>>>
>>> import inspect
>>> inspect.getargspec(my_wait.__call__)
ArgSpec(args=['self', 'attempt', 'delay'], varargs=None, keywords=None, defaults=None)
```

Reply:
Sorry I misread the line 89. I read `waiter = wait_base.__call__`. Facepalm.